File: swift/Sources/AOC2022/DayNN.swift
enum DayNN {
    static func solve1(_ input: String) throws -> Int {
        let lines = input.components(separatedBy: "\n")
        return try lines
            .map { l in try score1(l) }
            .reduce(0, +)
    }

    static func score1(_ line: String) throws -> Int {
        return 0
    }

    static func solve2(_ input: String) throws -> Int {
        let lines = input.components(separatedBy: "\n")
        return try lines
            .map { l in try score2(l) }
            .reduce(0, +)
    }

    static func score2(_ line: String) throws -> Int {
        return 0
    }
}

File: java/src/test/java/aoc2022/DayNN.java
package aoc2022;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class DayNN {

    public static long solve1(String input) {
        return input.lines()
                .mapToLong(DayNN::score1)
                .sum();
    }

    private static long score1(String s) {
        return 0;
    }

    public static long solve2(String input) {
        return input.lines()
                .mapToLong(DayNN::score2)
                .sum();
    }

    public static long score2(String s) {
        return 0;
    }

    @Test
    void example() {
        final String input = """
                                
                """;
        assertEquals(0, solve1(input));
        assertEquals(0, solve2(input));
    }

    @Test
    void input() throws Exception {
        var input = Resources.readString(Resources.class.getResource("/dayNN.txt"));
        assertEquals(0, solve1(input));
        assertEquals(0, solve2(input));
    }
}
